internal/provider/prior_utils.go

```diff
@@ -0,0 +1,21 @@
+package provider
+
+import "github.com/hashicorp/terraform-plugin-go/tfprotov5/tftypes"
+
+func extractStringFromPrior(prior map[string]tftypes.Value, key string) string {
```

Comment:
You shouldn't need to create these util functions in addition to the ones in `config_utils`.
`config`, `prior`, and `proposed` will always be the same data type.

Reply:
I removed this file and just used the config_utils lib instead.